Copy component docs (#1503)
hashicorp/design-system: website/docs/components/copy/snippet/partials/guidelines/guidelines.md

## Usage

### When to use

- To allow users to copy code snippets, such as URLs, tokens, IDs, etc.

### When not to use

- To copy content outside its container, use [Copy Button](/components/copy/button).
- In a code block, use [Copy Button](/components/copy/button).

### Effective Usage

> - There’s no limit to the number of `Copy::Snippet` components on a page, but maintain a balance to avoid overwhelming the user or cluttering the interface.

Reviewer: When speaking of the Ember component, I think saying "components" above is fine since there isn't a concept of a component "instance" in code like there is in Figma.

> - There’s no limit to the number of `Copy::Snippet` components on a page, but maintain a balance to avoid overwhelming the user or cluttering the interface.

Reviewer 2: Yep that's a good point!

> ### Effective Usage
> 
> - There’s no limit to the number of Copy Snippet instances on a page, but maintain a balance to avoid overwhelming the user or cluttering the interface.
> - Use the `Copy::Snippet` judiciously for crucial information that users frequently need. This keeps the copied content relevant and valuable without causing confusion or clutter.

Reviewer 3: ```suggestion
```

!!! Dont

> Take caution when using for full code snippets, instead consider using a more prominent code block with a [Copy Button](/components/copy/button).

Reviewer: I'd maybe just say "longer" code snippets for the above.

![Example of the copy snippet component not using proper usage] (/assets/components/copy/copy-snippet-dont.png)
!!!

## Color

> The `Copy::Snippet` comes in two colors: `primary` and `secondary`

Reviewer 4: We do often indicate property values within the guidelines, so I think this is fine, but I would update the examples to use a more "real-world" value instead of "Primary" and "Secondary". @andgen404 is there another example of proper usage of the primary variant from the audit? Couldn't hurt to add that example in as a "do" too.

<Doc::Layout @spacing="16px">
<Hds::Copy::Snippet @textToCopy="Primary" /> 
<Hds::Copy::Snippet @textToCopy="Secondary" @color="secondary" />
</Doc::Layout>

!!! Do

> When needing multiple `Copy::Snippets` in a single page, such as a Table, consider using the `secondary` color to reduce the prominence of each `Copy::Snippet`.

Reviewer: Maybe add:

"such as **_in_** a [Table]" (or "within") ?

## Usage

### When to use

> - To allow users to copy code snippets, such as URLs, tokens, IDs, etc.

Reviewer: I think someone said something during one of our meetings or in a comment somewhere about possibly not wanting to use the term "allow". (I know I suggested it here.)

So I was thinking that a better, more positive word choice would be "enable".

> - To allow users to copy code snippets, such as URLs, tokens, IDs, etc.

Reviewer: ```suggestion
- To enable users to copy code snippets, such as URLs, tokens, IDs, etc.
```

### When not to use

- To copy content outside its container, use [Copy Button](/components/copy/button).
- In a code block, use [Copy Button](/components/copy/button).

!!! Dont

Take caution when using for longer code snippets, instead consider using a more prominent code block with a [Copy Button](/components/copy/button).

> ![Example of the Copy Snippet with complex code] (/assets/components/copy/copy-snippet-dont.png)

Reviewer 2: ```suggestion
![Example of the Copy Snippet with complex code](/assets/components/copy/copy-snippet-dont.png)
```

!!!

## Color

The Copy Snippet comes in two colors: `primary` and `secondary`

<Doc::Layout @spacing="16px">
    <Hds::Copy::Snippet @textToCopy="Primary" /> 
    <Hds::Copy::Snippet @textToCopy="Secondary" @color="secondary" />
</Doc::Layout>

!!! Do

When multiple Copy Snippets are needed in a single page, such as in a [Table](/components/table), consider using the `secondary` color to reduce the prominence of each Copy Snippet.

![Example of the Copy Snippet component in a table](/assets/components/copy/copy-snippet-table-do.png)

!!!

!!! Do 

> When there are many secondary links on a page, like in a stepper form, it's important to use the `primary` color to highlight each Copy Snippet's importance for completing a task.

Reviewer: ```suggestion
When there are many secondary links on a page, like in a stepper form, it’s important to use the `primary` color to highlight each Copy Snippet’s importance for completing a task.
```

## Usage

### When to use

> - To allow users to copy code snippets, such as URLs, tokens, IDs, etc.

Reviewer 4: heh. I've seen you say this a few times, but it's a personal preference at this point. To my knowledge, we don't have any specific guidelines about moving away from this language. If you feel strongly that we should, we can discuss it.

## Usage

### When to use

> - Use the `CopySnippet` component when you want users to be able to copy code snippets or text, such as URLs, tokens, or IDs.

Reviewer: ```suggestion
- Use the `CopySnippet` component to allow users to copy code snippets or text, such as URLs, tokens, or IDs.
```

- Use when users need to extract and use specific information like API keys, configuration settings, or unique identifiers.

### When not to use

- Avoid using a `CopySnippet` component to copy content outside its container.

> - Don't use a `CopySnippet` in a code block. Instead, use a [Copy Button](/components/copy/button).

Reviewer: ```suggestion
- Don’t use a `CopySnippet` in a code block. Instead, use a [Copy Button](/components/copy/button).
```

### Effective Usage

> - There's no limit to the number of `CopySnippet` components on a page, but maintain a balance to avoid overwhelming the user or cluttering the interface.

Reviewer: ```suggestion
- There’s no limit to the number of `CopySnippet` components on a page, but maintain a balance to avoid overwhelming the user or cluttering the interface.
```

- Use the `CopySnippet` judiciously for crucial information that users frequently need. This keeps the copied content relevant and valuable without causing confusion or clutter.

!!! Dont

> Use the `CopySnippet` component for meaningful content that requires copying, while avoiding trivial or readily available text. Examples of suitable content include lengthy code snippets that don't need to be copied in their entirety and easily accessible text that users can manually copy at their discretion.

Reviewer: ```suggestion
Use the `CopySnippet` component for meaningful content that requires copying, while avoiding trivial or readily available text. Examples of suitable content include lengthy code snippets that don’t need to be copied in their entirety and easily accessible text that users can manually copy at their discretion.
```

![Example of the copy snippet component not using proper usage] (/assets/components/copy/copy-snippet-usage-example.png =675x*)

![Example of the copy snippet component not using proper usage] (/assets/components/copy/copy-snippet-usage-example-2.png =675x*)
!!!

## Table Usage
There are a few reasons why using a `CopySnippet` component in a table may not be ideal:

> 1. **Complexity and Readability**: Tables often contain organized data with multiple rows and columns. If the content within each cell is long or complicated, like code snippets or lengthy text, using the `CopySnippet` component may reduce readability and make it harder to understand the copied content within the table format.

Reviewer: ```suggestion
1. **Complexity and Readability**: Tables often contain organized data with multiple rows and columns. If the content within each cell is long or complicated, such as code snippets or lengthy text, using the `CopySnippet` component may reduce readability and make it harder to understand the copied content within the table format.
```
